tc-client/src/main/java/com/tc/object/DistributedObjectClient.java

```diff
@@ -407,15 +409,27 @@ private void openChannel(ClientMessageChannel channel) throws InterruptedExcepti
         DSO_LOGGER.error(handshake.getMessage());
         throw new IllegalStateException(handshake.getMessage(), handshake);
       } catch (final IOException ioe) {
-        DSO_LOGGER.debug("Unable to connect to server/s {} ...sleeping for 1 sec.", serverAddresses);
+        DSO_LOGGER.info("Unable to connect to server/s {} ...sleeping for {} sec.", serverAddresses, TimeUnit.MILLISECONDS.toSeconds(waitTime));
         DSO_LOGGER.debug("IOException connecting to server/s: {} {}", serverAddresses, ioe.getMessage());
         synchronized(clientStopped) {
-          clientStopped.wait(1000);
+          clientStopped.wait(waitTime);
         }
       }
     }
   }
-
+  
+  static long calculateWaitTime(long previousTime) {
```

Comment:
Considering the allowed ranges, I would instead implement that using an array of durations:
`1, 2, 3, 4, 6, 9, 14, 21, 30` and use a counter which is incremented